### ISSUE TITLE
IndexOutOfBoundsException when :concurrency is 1800+

### ISSUE THREAD

**Issue by ivos** [mhjort/clj-gatling#51]
Runs ok with :concurrency 1700 or less. With :concurrency 1800 or above throws on startup:

```
Exception in thread "main" java.lang.IndexOutOfBoundsException
	at clojure.lang.PersistentVector.assocN(PersistentVector.java:188)
	at clojure.lang.PersistentVector.assocN(PersistentVector.java:22)
	at clojure.lang.APersistentVector.assoc(APersistentVector.java:343)
	at clojure.lang.APersistentVector.assoc(APersistentVector.java:18)
	at clojure.lang.RT.assoc(RT.java:792)
	at clojure.core$assoc__4371.invokeStatic(core.clj:191)
	at clojure.core$update.invokeStatic(core.clj:5960)
	at clojure.core$update.invoke(core.clj:5952)
	at clj_gatling.simulation_util$split_to_buckets_with_sizes$fn__8097.invoke(simulation_util.clj:52)
	at clojure.lang.LongRange.reduce(LongRange.java:233)
	at clojure.core$reduce.invokeStatic(core.clj:6544)
	at clojure.core$reduce.invoke(core.clj:6527)
	at clj_gatling.simulation_util$split_to_buckets_with_sizes.invokeStatic(simulation_util.clj:51)
	at clj_gatling.simulation_util$split_to_buckets_with_sizes.invoke(simulation_util.clj:50)
	at clj_gatling.simulation_util$weighted_scenarios.invokeStatic(simulation_util.clj:77)
	at clj_gatling.simulation_util$weighted_scenarios.invoke(simulation_util.clj:68)
	at clj_gatling.simulation$run.invokeStatic(simulation.clj:153)
	at clj_gatling.simulation$run.invoke(simulation.clj:142)
	at clj_gatling.core$run.invokeStatic(core.clj:52)
	at clj_gatling.core$run.invoke(core.clj:44)
	at net.company.project.load.core$_main.invokeStatic(core.clj:24)
	at net.company.project.load.core$_main.invoke(core.clj:9)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at net.company.project.load.core.main(Unknown Source)
```

(Thanks for the Clojure wrapper around Gatling!)

**Comment by mhjort** [comment]
Apparently there is an issue. However, I was unable to reproduce this. It is probably related to the scenario structure you had. Could you pass the details of the simulation you were running. 

For example with this simulation defined in here: https://github.com/mhjort/clj-gatling-example/blob/master/src/clj_gatling_example/simulations.clj I was able to start it with concurrency 10000 (with my machine it is not possible to run properly with that kind of an concurrency but it does not crash)

**Comment by ivos** [comment]
There it is:

```
(ns net.company.project.load.simulations
  (:require [clojure.core.async :refer [chan go >!]]
            [org.httpkit.client :as http]
            ))

(defn create-pairing
  []
  {:name    "Customer: Create pairing"
   :request (fn [_]
              (let [response (chan)
                    process-result (fn [{:keys [status]}]
                                     (go
                                       (>! response [(and status (<= 200 status 299))])
                                       ))]
                (http/request {:method :post
                               :url    "http://localhost:8080/api/v1/pairings"}
                              process-result)
                response))})

(def purchase-simulation
  {:name      "Purchase simulation"
   :scenarios [
               {:name   "Happy purchase"
                :weight 100
                :steps  [
                         (create-pairing)
                         ]}
               {:name   "Canceled purchase"
                :weight 1
                :steps  [
                         (create-pairing)
                         ]}
               {:name   "Customer rejected purchase"
                :weight 1
                :steps  [
                         (create-pairing)
                         ]}
               {:name   "Balance rejected purchase"
                :weight 1
                :steps  [
                         (create-pairing)
                         ]}
               ]})

(def simulations
  {
   :purchase purchase-simulation
   })
```

I tried to cut down irrelevant things as much as possible. So e.g. those four "different" scenarios are meaningful, when I cut one out, the error goes away.

**Comment by mhjort** [comment]
Thanks. I can now also reproduce the problem. I will investigate this one.

**Comment by mhjort** [comment]
Found a root cause and managed to fix the bug. This is now fixed in `0.10.2`

**Comment by ivos** [comment]
I confirm it is resolved. Thank you.
